srml/timestamp/readme.md

```diff
@@ -0,0 +1,81 @@
+# Timestamp Module
+
+The timestamp module provides functionality to get and set the on-chain time.
+
+## Overview
+
+The timestamp module allows the validators to set and validate a timestamp with each block. It uses timestamp data as inherent data which is provided by the block author and validated/verified by other validators.
+
+The timestamp module is the recommended way to query the on-chain time instead of using an approach based on block numbers. The block numbers based time measurement can cause issues because of cummulative calculation errors and hence it should be avoided.
+
+## Public Interface
+
+### Supported origins
+
+* `inherent` - Used to set the timestamp for the current block
+
+### Types
+
+* `Moment` - Represents the current timestamp. Generally implemented as a `u64`.
+
+### Storage Items
+
+* `Now`: `Moment` - The current timestamp. A `u64` having the timestamps as **total seconds from the unix epoch**.
+* `BlockPeriod`: `Moment` - The minimum (and advised) period between blocks. Also represented as `Moment`.
+* `DidUpdate`: `bool` - Private flag to check if the timestamp was updated for the current block.
```

Comment:
General note, as these are private and implementation details, what is the benefit for the user?

Reply:
I think the idea of this section is so the user is at least aware of what is going in storage as a result of using this module. I guess we need to decide, is this so the user knows:

* What is going into storage, or
* What storage items are available from outside this module?

Reply:
We don't need to (and shouldn't) be documenting stuff that is internal to this module. Only interface is important.